android-components/components/browser/state/src/main/java/mozilla/components/browser/state/reducer/SearchReducer.kt

```diff
@@ -191,3 +192,9 @@ private fun BrowserState.updateDisabledSearchEngineIds(
         ),
     )
 }
+
+private fun BrowserState.restoreHiddenSearchEngines(): BrowserState {
+    return search.hiddenSearchEngines.fold(this) { state, engine ->
+        state.showSearchEngine(SearchAction.ShowSearchEngineAction(engine.id))
+    }
+}
```

Comment:
This confused me for a while, but I don't see any easy change that would make it clearer. We're iterating over the list of `hiddenSearchEngines` and then inside that, iterating over it again.

Why not just pull what's inside `showSearchEngine` out (without the `find` bit), and use that here? We're doing two loops here I think over the same list when we just need to pass in the `SearchEngine` to the `copy` function.

Does that make sense?